Clinical-Genomics/cg
# Upload MIP-RNA to all DNA cases in collaborator group matching by subject_ID

## Description
We want to upload all mip-RNA cases to DNA cases, matching on subject_id and tumour yes/no, to all customers within collaborator groups.

Linked issue: https://github.com/Clinical-Genomics/project-planning/issues/316

### Added

-

### Changed

- Query to upload RNA cases matches a list of customer IDs instead of only the same customer ID.

### Fixed

- Function `get_samples_by_customer_id_list_and_subject_id_and_is_tumour` did not check tumour state, only tumour=True, it now checks tumour state from the input.


### How to prepare for test
We are doing the full test. 
1) On stage cg master branch: A MIP-RNA trio (smoothtuna), which is going to be uploaded to two MIP-DNA trios of the same customer (onkiwi, stablecorgi). 
2) Then we change customer of stablecorgi, which should then only upload to onkiwi. 
3) Install this PR branch, and do the upload again as described in test 2. It should upload to both customers.

- [ ] Ssh to hasta
- [ ] Install **master** on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b master -a
    ```
- [ ]  get the associated DNA files to stage for onkiwi `mkdir -p /home/proj/stage/housekeeper-bundles/onkiwi/2021-12-24/` and `cp -r /home/proj/production/housekeeper-bundles/onkiwi/2021-12-24/ /home/proj/stage/housekeeper-bundles/onkiwi/2021-12-24/`
- [ ] do `cg upload -f onkiwi`
- [ ] `mkdir /home/proj/stage/housekeeper-bundles/stablecorgi/2022-12-15/` and `cp -r /home/proj/production/housekeeper-bundles/stablecorgi/2022-12-15/ /home/proj/stage/housekeeper-bundles/stablecorgi/2022-12-15/`
- [ ] do `cg upload -f stablecorgi`
- [ ] get the associated RNA files to stage `mkdir -p /home/proj/stage/housekeeper-bundles/smoothtuna/2022-09-02/` and `cp -r /home/proj/production/housekeeper-bundles/smoothtuna/2022-09-02/ /home/proj/stage/housekeeper-bundles/smoothtuna/2022-09-02/ and cp /home/proj/production/housekeeper-bundles/smoothtuna/2022-09-02/smoothtuna_report.pdf /home/proj/stage/housekeeper-bundles/smoothtuna/2022-09-02/`

### How to test
Subject_id C000357840 & C000357838 have only 1 DNA and 1 RNA sample connected. SubjectID C000350464 has 5 samples connected, of which 3 RNA and 2 DNA. Removed subject ID C000350464 from DNA sample ACC10362A7, to make it 1 DNA sample connected. The extra RNA samples should be not be relevant for the upload.

#### Test 1, upload as normal on master.
- [ ] Do `cg upload rna-to-scout smoothtuna`

#### Test 2, upload on master after changing customers in statusDB stage for case stablecorgi.
- [ ] Change customer in statusDB stage for stablecorgi from 3 to 4.
- [ ] Do `cg upload rna-to-scout smoothtuna`

#### Test 3, install this PR branch, and do the upload again.
- [ ] Install this branch:
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b upload-mip-rna-to-collaborators -a
    ```
- [ ] Do `cg upload rna-to-scout smoothtuna`

### Expected test outcome
#### Test 1, upload as normal on master.
- [x] Check that all relevant files are uploaded to 2 DNA cases.
<img width="1230" alt="image" src="https://user-images.githubusercontent.com/57712924/235093378-dbc9cc7b-0153-4a70-8642-d039a72346a4.png">

#### Test 2, only 1 DNA case connected for the relevant customer.
- [x] Check that all relevant files are uploaded to onkiwi only.
<img width="1228" alt="image" src="https://user-images.githubusercontent.com/57712924/235095924-f5a9be93-872a-451b-a280-d58dd9020ede.png">

#### Test 3, RNA case is uploaded to 2 DNA cases, each being from a different customer.
- [x] Check that all relevant files are uploaded to 2 DNA cases.
<img width="1231" alt="image" src="https://user-images.githubusercontent.com/57712924/235100811-2d21086b-a252-4c1d-8d12-aacdfcdadd1e.png">


## Review

Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a


## Implementation Plan
